### PR TITLE
Expand bonk damage options

### PR DESCRIPTION
**Pull request by mracsys**
There have been multiple requests to refine bonk damage beyond either nothing or one bonk kills. This PR allows customizing bonk damage similar to damage multiplier settings. All damage values except one bonk KO respect double defense.

### CODE REVIEW

File: data/LogicHelpers.json [hunk 1]
```diff
@@ -61,9 +61,9 @@
             or (for_age == adult and (Bow or Hookshot))
             or (for_age == both and (Slingshot or Boomerang) and (Bow or Hookshot))
             or (for_age == either and (Slingshot or Boomerang or Bow or Hookshot))",
-    "can_bonk_tree": "deadly_bonks == False or Fairy or can_use(Nayrus_Love)",
-    "can_bonk_crate": "deadly_bonks == False or Fairy or can_use(Nayrus_Love) or can_blast_or_smash",
-    "can_bonk_underwater_crate": "deadly_bonks == False or Fairy or can_use(Nayrus_Love)",
+    "can_bonk_tree": "deadly_bonks == 'ohko' or Fairy or can_use(Nayrus_Love)",
+    "can_bonk_crate": "deadly_bonks == 'ohko' or Fairy or can_use(Nayrus_Love) or can_blast_or_smash",
+    "can_bonk_underwater_crate": "deadly_bonks == 'ohko' or Fairy or can_use(Nayrus_Love)",
```
Comment (Cuphat) [inline data/LogicHelpers.json:66]: Shouldn't these be `deadly_bonks != 'ohko'`?

File: SettingsList.py [hunk 2]
```diff
@@ -4511,15 +4511,23 @@ def __init__(self, name, gui_text, min, max, default, step=1,
...
+            'half':      'Quarter Heart',
+            'normal':    'Half Heart',
+            'double':    'Whole Heart',
+            'quadruple': 'Two Hearts',
+            'ohko':      'One Bonk KO',
+        },
         gui_tooltip    = '''\
             When rolling, hitting a wall or object
             will kill Link.
         ''',
```
Comment (Cuphat) [inline SettingsList.py:4529]: Probably needs a description update.

**Review by Cuphat — CHANGES_REQUESTED** [review]

File: data/LogicHelpers.json (hunk 1 above)
Comment (mracsys) [reply data/LogicHelpers.json:66]: yes, they should

**Review by Cuphat — APPROVED** [review]

**Comment by ETR-BTF** [comment]
How does this setting interact with the normal damage multiplier setting?

**Comment by mracsys** [comment]
Multipliers are separate. Since the GUI text specifies how many hearts are lost on bonking it felt weird to make it differ from what was requested, or for OHKO to automatically enable bonko if bonk damage is nonzero.

**Comment by mracsys** [comment]
There is a bug with the two heart setting where damage is not applied. Taking to draft until it is resolved.

**Comment by mracsys** [comment]
Bug addressed, somehow screwed up copy/paste from damage multipliers for just two hearts/"quad".

**Review by cjohnson57 — APPROVED** [review]

**Comment by r0bd0g** [comment]
I don't think the names of this setting make sense since "Normal" bonk damage is 0.

**Pull request merged**